source/hp/dof_handler.cc

```diff
@@ -1506,6 +1506,30 @@ namespace hp
 
 
 
+  template <int dim, int spacedim>
+  void
+  DoFHandler<dim, spacedim>::execute_coarsening_and_refinement()
```

Comment:
Perhaps we should guide users a bit with this function.

what I did in `tests/hp/laplace.h` for https://github.com/dealii/dealii/pull/7513  was the following:

1. Mark cells for h-refinement using standard tools
2.selectively substituted some h-refinement flags for p-refinement
call triangulation.prepare_coarsening_and_refinement();
3. setup Solution Transfer and call
  soltrans.prepare_for_coarsening_and_refinement(solution_coarse);

4. **** increment fe_index *****
5. execute Refinement
  triangulation.execute_coarsening_and_refinement();
6. setup_system();
7. interpolate via solution transfer
  soltrans.interpolate(solution_coarse, solution);

Given that the sequence is not trivial in the presence of solution transfer, perhaps we can briefly document this in the function description. Maybe say something like

> If the solution transfer is to be done in the hp-FEM case, this function should be called after initialization of the solution transfer object and before calling `execute_coarsening_and_refinement()`.

Reply:
Agreed. I'll extend the documentation for this function.

I guess a good place to give users a general overview over the functionality would be `doc/doxygen/headers/hp.h`. However, I would postpone that until the day we're happy with the interface (i.e. #7515 is done).